### PR TITLE
Update to otel-java-instrumentation 1.0.0

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**

### CODE REVIEW

**Review by iNikem — APPROVED** [review]

**Review by jkwatson — APPROVED** [review]

File: instrumentation/netty-4.0/src/test/java/com/splunk/opentelemetry/netty/v4_0/NettyInstrumentationTest.java [hunk 1]
```diff
@@ -120,15 +124,9 @@ private static void assertHeaders(Response response, String serverTimingHeader)
...
-
-    var traces = exporter.getTraces();
-    assertEquals(1, traces.size());
-
-    var spans = traces.get(0);
-    assertEquals(1, spans.size());
+    instrumentation.waitForTraces(1);
 
-    var serverSpan = spans.get(0);
+    var serverSpan = instrumentation.spans().get(0);
```
Comment (breedx-splk) [inline instrumentation/netty-4.0/src/test/java/com/splunk/opentelemetry/netty/v4_0/NettyInstrumentationTest.java:129]: Will this fail if 1000 traces come back?  Seems like we were checking explicitly before but are not now...?

File: instrumentation/servlet-3-testing/src/test/java/com/splunk/opentelemetry/servlet/Servlet3InstrumentationTest.java [hunk 2]
```diff
@@ -135,15 +132,9 @@ private static void assertHeaders(Response response, String serverTimingHeader)
...
-
-    var traces = exporter.getTraces();
-    assertEquals(1, traces.size());
-
-    var spans = traces.get(0);
-    assertEquals(1, spans.size());
+    instrumentation.waitForTraces(1);
 
-    var serverSpan = spans.get(0);
+    var serverSpan = instrumentation.spans().get(0);
```
Comment (breedx-splk) [inline instrumentation/servlet-3-testing/src/test/java/com/splunk/opentelemetry/servlet/Servlet3InstrumentationTest.java:137]: same question about the number of traces also being validated.

File: instrumentation/servlet/src/test/java/com/splunk/opentelemetry/servlet/ServletInstrumentationTest.java [hunk 3]
```diff
@@ -134,15 +131,9 @@ private static void assertHeaders(Response response, String serverTimingHeader)
...
-
-    var traces = exporter.getTraces();
-    assertEquals(1, traces.size());
-
-    var spans = traces.get(0);
-    assertEquals(1, spans.size());
+    instrumentation.waitForTraces(1);
 
-    var serverSpan = spans.get(0);
+    var serverSpan = instrumentation.spans().get(0);
```
Comment (breedx-splk) [inline instrumentation/servlet/src/test/java/com/splunk/opentelemetry/servlet/ServletInstrumentationTest.java:136]: same question about the number of traces being validated.

**Review by breedx-splk — APPROVED** [review]
Looks good to me, just had a question about the tests changing (to make sure we aren't losing coverage).

File: instrumentation/netty-4.0/src/test/java/com/splunk/opentelemetry/netty/v4_0/NettyInstrumentationTest.java (hunk 1 above)
Comment (mateuszrzeszutek) [reply instrumentation/netty-4.0/src/test/java/com/splunk/opentelemetry/netty/v4_0/NettyInstrumentationTest.java:129]: Hmm, you're right. I'll revert this code to assert trace/span counts

**Comment by mateuszrzeszutek** [comment]
This PR will need to wait for 0.17.1

**Comment by mateuszrzeszutek** [comment]
Finally updated to 1.0.0!

@laurit can you confirm that the app server smoke test span names are OK? Thanks!

File: build.gradle [hunk 4]
```diff
@@ -15,11 +15,12 @@ group = "com.splunk.public"
...
 def versions = [
-    'opentelemetry'         : snapshot ? "0.17.0-SNAPSHOT"       : "0.16.0",
-    'opentelemetryAlpha'    : snapshot ? "0.17.0-alpha-SNAPSHOT" : "0.16.0-alpha",
-    'opentelemetryJavaagent': snapshot ? "0.17.0-SNAPSHOT"       : '0.16.1',
-    'mockito'               : '3.8.0',
-    'jupiter'               : '5.7.1'
+    'opentelemetry'              : snapshot ? "0.18.0-SNAPSHOT"       : "1.0.0",
+    'opentelemetryAlpha'         : snapshot ? "0.18.0-alpha-SNAPSHOT" : "1.0.0-alpha",
+    'opentelemetryJavaagent'     : snapshot ? "0.18.0-SNAPSHOT"       : '1.0.0',
+    'opentelemetryJavaagentAlpha': snapshot ? "0.18.0-alpha-SNAPSHOT" : '1.0.0-alpha',
```
Comment (iNikem) [inline build.gradle:21]: Certainly we already have 1.0.0-SNAPSHOT?

**Review by iNikem — APPROVED** [review]

**Review by laurit — APPROVED** [review]

File: build.gradle (hunk 4 above)
Comment (mateuszrzeszutek) [reply build.gradle]: Fixed!

**Pull request merged**